src/include/storage/index/hash_index.h

```diff
@@ -184,7 +219,8 @@ class HashIndex final : public OnDiskHashIndex {
     std::unique_ptr<BaseDiskArray<Slot<T>>> pSlots;
     std::unique_ptr<BaseDiskArray<Slot<T>>> oSlots;
     OverflowFileHandle* overflowFileHandle;
-    std::unique_ptr<HashIndexLocalStorage<LocalStorageType>> localStorage;
+    std::unique_ptr<HashIndexLocalStorage<BufferKeyType>> localStorage;
+    HashIndexBuilder<T> bulkInsertLocalStorage;
```

Comment:
When I'd benchmarked it before it was somewhat slower, but I'm also planning on trying to optimize it somewhat now that we've removed the constraint that it must be able to be directly written to disk. I imagine that after removing the constraint on the number of primary slots for `bulkInsertLocalStorage` it will be close enough that the difference is dominated by the cost of writing to the WAL anyway.